docs/rfcs/016-user-and-database-management.md

```diff
@@ -0,0 +1,70 @@
+## Zentih user and database management RFC
+
+### CREATE USER
+
+CREATE USER is only allowed from console, 
+because console is involved in the auth process, when users connect via proxy.
+It's much handier to have a single source of truth.
+
+Client has to chose which databases will be accessible by the new user.
+Along with database access, we explicitly grant CREATE ON DATABASE privilege to the user.
+`GRANT CREATE ON DATABASE .. TO .. ;`
+to let the user to create extensions.
+
+All new users are granted with CREATEDB privilege by default.
+`CREATE ROLE .. WITH CREATEDB`
+Later we may add an option to manage it.
+
+For now we do not allow clients to create new users and NOLIGIN roles to manage privileges.
+i.e. `CREATE ROLE .. WITH CREATEROLE`
+Let's postpone this until we got N feature requests about more complex roles management.
+
+
+### CREATE DATABASE
+
+Client can CREATE DATABASE via SQL as well as via console.
+
+When database is created from SQL (not by superuser)
+we intercept this action using DDL hook and send request to the console to synchronize the information about databases.
```

Comment:
We have a UI to query the compute, and you can select some particular user / db pair from existing ones there. It's just much better UX, when you don't need to type in user and db names, and can use GUI selector instead

Reply:
I, too, do not understand why Console needs to manage databases.

Reply:
Ain't that an answer above? Manage --- likely not, at least not a single entry point for creating them. Know about them --- definitely yes, because of the reason above.

Reply:
Sorry, my tab hadn't yet synchronized your reply when I opened the issue, and the tab also didn't receive your reply until I sent my comment.

Either way, can't we query the catalogs of the clusters directly? That might be a bad idea, but at the same time, it shouldn't be that hard -- the database's OID is part of each segment, and thus we should be able to extract that information quite easily without specifically involving the console. Console might need to call into PageServer for this information, but that's already part of its job.

Reply:
> If we use the DDL hook here then what happens if one creates a database during a console outage? Is it possible for DDL not to fail the database creation during console outage and silently finish?
> And we should probably have a background routine scanning databases anyway, because the console may be offline during creation.

Yes, that was my understanding to. We don't do any blocking operations in the DDL hook. It's just an optional notify to speed up syncing. Console should anyway scan / sync scheme from time to time. E.g. we can do it before suspend and/or during availability checks for active clusters (not implemented yet).